src/crimson/osd/main.cc

```diff
@@ -114,7 +114,9 @@ int main(int argc, char* argv[])
     ("mkfs", "create a [new] data directory")
     ("debug", "enable debug output on all loggers");
 
-  auto [ceph_args, app_args] = partition_args(app, argv, argv + argc);
+  std::vector<const char*> ceph_args, app_args;
+  std::tie(ceph_args, app_args) = partition_args(app, argv, argv + argc);
+
```

Comment:
instead, i'd recommend following patch
```diff
diff --git a/src/crimson/osd/main.cc b/src/crimson/osd/main.cc
index 8d9152debd..4ffe645468 100644
--- a/src/crimson/osd/main.cc
+++ b/src/crimson/osd/main.cc
@@ -134,7 +134,8 @@ int main(int argc, char* argv[])
   using crimson::common::sharded_conf;
   using crimson::common::sharded_perf_coll;
   try {
-    return app.run_deprecated(app_args.size(), const_cast<char**>(app_args.data()), [&] {
+    return app.run_deprecated(app_args.size(), const_cast<char**>(app_args.data()),
+      [&, &ceph_args=ceph_args] {
       auto& config = app.configuration();
       return seastar::async([&] {
    if (config.count("debug")) {
```

Reply:
WIll post a new version with this change